metpy/tests/test_xarray.py

```diff
@@ -124,3 +141,237 @@ def test_strftime():
+
+    # Must compare both values and attributes
+    assert (x == test_var['x']).all() and x.attrs == test_var['x'].attrs
+    assert (y == test_var['y']).all() and y.attrs == test_var['y'].attrs
+    assert ((vertical == test_var['isobaric']).all() and
+            vertical.attrs == test_var['isobaric'].attrs)
+    assert (time == test_var['time']).all() and time.attrs == test_var['time'].attrs
+
+
+def test_coordinates_basic_by_property(test_var):
+    """Test that NARR example coordinates are like we expect using properties."""
+    # Must compare both values and attributes
+    assert ((test_var.metpy.x == test_var['x']).all() and
+            test_var.metpy.x.attrs == test_var['x'].attrs)
+    assert ((test_var.metpy.y == test_var['y']).all() and
+            test_var.metpy.y.attrs == test_var['y'].attrs)
+    assert ((test_var.metpy.vertical == test_var['isobaric']).all() and
+            test_var.metpy.vertical.attrs == test_var['isobaric'].attrs)
+    assert ((test_var.metpy.time == test_var['time']).all() and
+            test_var.metpy.time.attrs == test_var['time'].attrs)
+
+
+def test_coordinates_specified_by_name_with_dataset(test_ds_generic):
+    """Test that we can manually specify the coordinates by name."""
+    data = test_ds_generic.metpy.parse_cf(coordinates={'T': 'a', 'Z': 'b', 'Y': 'c', 'X': 'd'})
+    x, y, vertical, time = data['test'].metpy.coordinates('x', 'y', 'vertical', 'time')
+
+    # Must compare both values and attributes
+    assert (x == data['test']['d']).all() and x.attrs == data['test']['d'].attrs
+    assert (y == data['test']['c']).all() and y.attrs == data['test']['c'].attrs
+    assert (vertical == data['test']['b']).all() and vertical.attrs == data['test']['b'].attrs
+    assert (time == data['test']['a']).all() and time.attrs == data['test']['a'].attrs
+
+
+def test_coordinates_specified_by_dataarray_with_dataset(test_ds_generic):
+    """Test that we can manually specify the coordinates by DataArray."""
+    data = test_ds_generic.metpy.parse_cf(coordinates={
+        'T': test_ds_generic['d'],
+        'Z': test_ds_generic['a'],
+        'Y': test_ds_generic['b'],
+        'X': test_ds_generic['c']
+    })
+    x, y, vertical, time = data['test'].metpy.coordinates('x', 'y', 'vertical', 'time')
+
+    # Must compare both values and attributes
+    assert (x == data['test']['c']).all() and x.attrs == data['test']['c'].attrs
+    assert (y == data['test']['b']).all() and y.attrs == data['test']['b'].attrs
+    assert (vertical == data['test']['a']).all() and vertical.attrs == data['test']['a'].attrs
+    assert (time == data['test']['d']).all() and time.attrs == data['test']['d'].attrs
+
+
+def test_bad_coordinate_type(test_var):
+    """Test that an AttributeError is raised when a bad axis/coordinate type is given."""
+    with pytest.raises(AttributeError) as exc:
+        axis, = test_var.metpy.coordinates('bad_axis_type')
```

Comment:
You can just call the function here, no need to do anything with the return value, so just:
```python
test_var.metpy.coordinates('bad_axis_type')
```

Reply:
I had tried that, but because the `coordinates` method is a generator function, I needed to assign it in order for the underlying function to run and raise the error. Is there a better way around this?

Reply:
Ah, generator. You could wrap the call in `next(...)`.